swarm/src/lib.rs

```diff
@@ -1583,27 +1583,36 @@ mod tests {
         TBehaviour: NetworkBehaviour,
         <<TBehaviour::ConnectionHandler as IntoConnectionHandler>::Handler as ConnectionHandler>::OutEvent: Clone,
     {
-        [swarm1, swarm2]
-            .iter()
-            .all(|s| s.behaviour.inject_connection_established.len() == num_connections)
```

Comment:
Counting the `inject_connection_established` calls only might mislead in case there are more than 2 swarms interconnecting. Also some of these connections might already be closed.